Source/WebGPU/WebGPU/CommandEncoder.swift

```diff
@@ -32,7 +38,617 @@ public func resolveQuerySet(commandEncoder: WebGPU.CommandEncoder, querySet: Web
+                // https://developer.apple.com/documentation/metal/mtlblitcommandencoder/1400756-copyfromtexture?language=objc
+                // "When you copy to a 1D texture, height and depth must be 1."
+                let sourceSize = MTLSizeMake(Int(widthForMetal), 1, 1)
+                let sourceOrigin = MTLOriginMake(Int(source.origin.x), 0, 0)
+                for layer in 0...copySize.depthOrArrayLayers {
+                    var layerTimesDestinationBytesPerImage = UInt(layer)
+                    (layerTimesDestinationBytesPerImage, didOverflow) = layerTimesDestinationBytesPerImage.multipliedReportingOverflow(by: destinationBytesPerImage)
+                    guard !didOverflow else {
+                        return
+                    }
+                    var destinationOffset = UInt(destination.layout.offset)
+                    (destinationOffset, didOverflow) = destinationOffset.addingReportingOverflow(layerTimesDestinationBytesPerImage)
+                    guard !didOverflow else {
+                        return
+                    }
+                    var sourceSlice = UInt(source.origin.z)
+                    (sourceSlice, didOverflow) = sourceSlice.addingReportingOverflow(UInt(layer))
+                    guard !didOverflow else {
+                        return
+                    }
+                    var widthTimesBlockSize = UInt(widthForMetal)
+                    (widthTimesBlockSize, didOverflow) = widthTimesBlockSize.multipliedReportingOverflow(by: blockSize.value())
+                    guard !didOverflow else {
+                        return
+                    }
+                    let sum = UInt(destinationOffset) + UInt(widthTimesBlockSize)
+                    if sum > destinationBuffer.length {
+                        continue
+                    }
+                    blitCommandEncoder.copy(
+                        from: sourceTexture.texture(),
+                        sourceSlice: Int(sourceSlice),
+                        sourceLevel: Int(source.mipLevel),
+                        sourceOrigin: sourceOrigin,
+                        sourceSize: sourceSize,
+                        to: destinationBuffer,
+                        destinationOffset: Int(destinationOffset),
+                        destinationBytesPerRow: Int(destinationBytesPerRow),
+                        destinationBytesPerImage: Int(destinationBytesPerImage),
+                        options: options)
+                }
+            case WGPUTextureDimension_2D:
+                // https://developer.apple.com/documentation/metal/mtlblitcommandencoder/1400756-copyfromtexture?language=objc
+                // "When you copy to a 2D texture, depth must be 1."
+                let sourceSize = MTLSizeMake(Int(widthForMetal), Int(heightForMetal), 1)
+                let sourceOrigin = MTLOriginMake(Int(source.origin.x), Int(source.origin.y), 0)
+                for layer in 0...copySize.depthOrArrayLayers {
```

Comment:
This is off by one. The old code said

`        for (uint32_t layer = 0; layer < copySize.depthOrArrayLayers; ++layer) {`

You need `..<`